addons/mail/static/src/core/common/message_service.js

```diff
@@ -30,6 +30,9 @@ export class MessageService {
         if (convertBrToLineBreak(message.body) === body && attachments.length === 0) {
             return;
         }
+        await Promise.all(message.mentionedChannelPromises).then((fetchedChannels) => {
+            mentionedChannels.add(...fetchedChannels);
+        });
```

Comment:
Doesn't it crash when invalid channel are given? As they are not filtered out unless I'm reading this incorrectly.